src/main/scala/strawman/collection/mutable/ArrayBuffer.scala

```diff
@@ -78,16 +121,46 @@ object ArrayBuffer extends IterableFactory[ArrayBuffer] {
   /** Avoid reallocation of buffer if length is known. */
   def fromIterable[B](coll: Iterable[B]): ArrayBuffer[B] =
     if (coll.knownSize >= 0) {
-      val elems = new Array[AnyRef](coll.knownSize)
+      val array = new Array[AnyRef](coll.knownSize)
       val it = coll.iterator()
-      for (i <- 0 until elems.length) elems(i) = it.next().asInstanceOf[AnyRef]
-      new ArrayBuffer[B](elems, elems.length)
+      for (i <- 0 until array.length) array(i) = it.next().asInstanceOf[AnyRef]
+      new ArrayBuffer[B](array, array.length)
     }
     else new ArrayBuffer[B] ++= coll
 }
 
-class ArrayBufferView[A](val elems: Array[AnyRef], val start: Int, val end: Int) extends IndexedView[A] {
-  def length = end - start
-  def apply(n: Int) = elems(start + n).asInstanceOf[A]
+class ArrayBufferView[A](val array: Array[AnyRef], val length: Int) extends IndexedView[A] {
+  def apply(n: Int) = array(n).asInstanceOf[A]
   override def className = "ArrayBufferView"
 }
+
+/** An object used internally by collections backed by an extensible Array[AnyRef] */
+object RefArrayUtils {
+
+  def ensureSize(array: Array[AnyRef], end: Int, n: Int): Array[AnyRef] = {
+    // Use a Long to prevent overflows
+    val arrayLength: Long = array.length
+    def growArray = {
+      var newSize: Long = arrayLength * 2
+      while (n > newSize)
+        newSize = newSize * 2
+      // Clamp newSize to Int.MaxValue
+      if (newSize > Int.MaxValue) newSize = Int.MaxValue
+
+      val newArray: Array[AnyRef] = new Array(newSize.toInt)
+      Array.copy(array, 0, newArray, 0, end)
+      newArray
+    }
+    if (n <= arrayLength) array else growArray
+  }
+
+  /** Remove elements of this array at indices after `sz`.
+   */
+  def nullElems(array: Array[AnyRef], start: Int, end: Int): Unit = {
+    var i = start
+    while (i < end) {
+      array(i) = null
+      i += 1
+    }
+  }
```

Comment:
Except for very short arrays, this would be faster as `java.util.Arrays.fill(array, start, end, null)`